InteractionBase/src/vis/ViewSwitcherMenu.cpp

```diff
@@ -41,11 +41,17 @@ void ViewSwitcherMenu::show(Visualization::Item* target)
 
 void ViewSwitcherMenu::showNow(Visualization::Item* target)
 {
-	QVector<Visualization::Item*> items;
-	for (auto view : target->scene()->viewItems())
-		items.append(new VViewSwitcherEntry(nullptr, view->name()));
-	for (int i = items.size(); i < 9; i++)
-		items.append(new VViewSwitcherEntry(nullptr, "Empty slot " + QString::number(i)));
+	QVector<QVector<Visualization::Item*>> items;
+	items.resize(target->scene()->viewItems().size());
+	for (int col = 0; col < target->scene()->viewItems().size(); col++)
+		for (int row = 0; row < target->scene()->viewItems()[col].size(); row++)
+			if (target->scene()->viewItems()[col][row])
+				items[col].append(new VViewSwitcherEntry(nullptr, target->scene()->viewItems()[col][row]->name()));
+			else
+				items[col].append(new VViewSwitcherEntry(nullptr, "Empty slot"));
+	for (int col = 0; col < items.size(); col++)
+		for (int row = items[col].size(); row < 3; row++)
+			items[col].append(new VViewSwitcherEntry(nullptr, "Empty slot"));
 	Menu::hideNow();
 	Menu::instance = new ViewSwitcherMenu(items, target);
```

Comment:
Now that you're passing a 2d vector, could you please remove the constructor of `ViewSwitcherMenu` that takes a 1d vector/list? There was also an associated `arrange` method, which you should also remove.